web/src/server/auth.ts

```diff
@@ -264,6 +280,19 @@ const extendedPrismaAdapter: Adapter = {
 
     return user;
   },
+
+  async linkAccount(data) {
```

Comment:
I'm trying to understand here how the code works, so sorry in advance if the questions are obvious (newb here in TS).

In line 259 the `extendedPrismaAdapter` is built which has all the methods of the original `prismaAdapter` (line 260). Then here we overwrite the `linkAccount` method where we `delete` the Keycloak extra fields before calling the original `linkAccount`. This will happen to all calls to `extendedPrismaAdapter.linkAccount`, correct? If yes, I assume also calls when other providers are used. If that happens, is it ok to `delete` an inexistent field? Are there other "interferences" we would face when using different providers at the same time?

In the languages I'm more familiar with one way to do this would be building a different `extendedPrismaAdapter` class for each different type of provider, all inheriting from the same interface or class, and each one overriding (if necessary) the parent's methods (in this case `linkAccount`).

Just to be clear, I don't think this is a bad design. I'm just trying to understand how such things are done in Typescript!

Reply:
Ah! I see! 😄 Well I think this is a trade-off of complexity. In my opinion, since `delete` of inexistent fields is safe (and maybe idiomatic) in JS, I would keep it as your initial proposal. lgtm

I am not a core maintainer so I can't merge this PR. I think we need @marcklingen for that.

Reply:
@mparada-suva Thank you for your suggestion.